src/App.js

```diff
@@ -1,26 +1,45 @@
-import React from 'react';
-import logo from './logo.svg';
+import React, { Component } from 'react'
 import './App.css';
+import Home from './components/home/Home';
+import { Route, Switch } from 'react-router-dom'
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+import RandomBeer from './components/randombeer/RandomBeer';
+import NewBeer from './components/newbeer/NewBeer';
+import Header from './components/header/Header';
+import AllBeers from './components/allbeers/AllBeers';
+import BeerDetail from './components/beerdetail/BeerDetail';
+
+
+export default class App extends Component {
+
+  state = {
+    beers: ''
+  }
+
+  updateStateAppJs = (beersFromDB) => {
+    this.setState({beers: beersFromDB})
+  }
+
+  render() {
+
+    return (
+      <div>
+      <Route path='/beer' component={Header} />
+    
+     <Switch>
+       <Route exact path='/' component={Home}/>
+       <Route exact path='/beer/beers' render= {() => <AllBeers  updateStateAppJs={this.updateStateAppJs}/>}/>
+       <Route exact path='/beer/beers/:idBeer' render={(props) => <BeerDetail {...props} beers={this.state.beers} /> }/>
```

Comment:
The BeerDetail component needs to access just a single beer object. Is not needed to send all the objects in the array